drivers/clock_control/clock_control_ra2_hoco_osc.c

```diff
@@ -0,0 +1,130 @@
+/**
+ * clock_control_ra2_hoco_osc.c
+ *
+ * RA2 High-speed on-chip oscillator (HOCO) driver implementation
+ *
+ * Copyright (c) 2022-2024 MUNIC Car Data
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/irq.h>
+#include "clock_control_ra2_priv.h"
+
+#define DT_DRV_COMPAT	renesas_ra2_hoco_osc
+#define HOCO_NODE	DT_DRV_INST(0)
+
+struct ra_hoco_osc_config {
+	/* Must be first */
+	struct ra_common_osc_config common;
+};
+
+static
```

Comment:
we don't put static on a different line in zephyr, goes on the same line

Reply:
Actually please just clang-format all of these files using the style included with zephyr

Reply:
> Actually please just clang-format all of these files using the style included with zephyr

No so easy. Unfortunately clang-format adore to break complex defs.

Reply:
> we don't put static on a different line in zephyr, goes on the same line

I can fix certainly the kernel's style "static" .

However, wouldn't be better to update following sentence in Zephyr's docos before?

From here https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style:
>
>...
>In general, follow the [Linux kernel coding style](https://kernel.org/doc/html/latest/process/coding-style.html), with the following exceptions:
>...
> When there are differences between the [Coding Style](https://docs.zephyrproject.org/latest/contribute/guidelines.html#id2) guidelines and the formatting generated by code formatting tools, the [Coding Style](https://docs.zephyrproject.org/latest/contribute/guidelines.html#id2) guidelines take precedence. If there is ambiguity between formatting tools and the guidelines, maintainers may decide which style should be adopted.

Because the LCG doesn't prohibit placing `static` on different line, and I didn't find such rule in the Zephyr's coding style guideline too.